### ISSUE TITLE
RegisterForReflection#classNames does not register full hierarchy of the class

### ISSUE THREAD

**Issue by radcortez** [quarkusio/quarkus#21985]
### Description

If we use `io.quarkus.runtime.annotations.RegisterForReflection#classNames` to register classes for reflection it does not register classes up in the hierarchy.

On the other hand, if we use `-H:ReflectionConfigurationFiles`, it does register the full class hierarchy.

### Implementation ideas

I think we should look for the full hierarchy here: https://github.com/quarkusio/quarkus/blob/c82f416fd1ad3ddf8ee40cd40c839e9c884757b3/core/deployment/src/main/java/io/quarkus/deployment/steps/RegisterForReflectionBuildStep.java#L66

And also register super classes.

Are we ok to implement this? I can do it.

**Comment by geoand** [comment]
I don't think we want to do this TBH since it could inflate our native images a lot. IMHO, we should just keep using `ReflectiveClassBuildItem` and `ReflectiveHierarchyBuildItem`

**Comment by geoand** [comment]
Yeah, could do that I guess

**Comment by radcortez** [comment]
Not a big deal anyway, if no one else complained yet :)

**Comment by geoand** [comment]
:)

**Comment by mkouba** [comment]
> Maybe we could add a new parameter into the annotation to flag if the entire hierarchy of the class should be registered?

+1

**Comment by aadityasinha-dotcom** [comment]
Is this issue open?

**Comment by gsmet** [comment]
@aadityasinha-dotcom sure!

**Comment by aadityasinha-dotcom** [comment]
Can you explain me about this issue?

**Comment by igorregis** [comment]
Hi, I would like to sign up for this issue. I have been doing some tests and I guess I got the point to correctly implement it.

**Comment by igorregis** [comment]
I've got the fix done, but not in a way I would like to. The reason is the AnnotationInstance misbehavior. 

The getBooleanValue method (below) of the class RegisterForReflectionBuildStep uses it to get the value of the annotation. 
https://github.com/quarkusio/quarkus/blob/fe7332e0585e4f83b674357857369fe28cbcbbdf/core/deployment/src/main/java/io/quarkus/deployment/steps/RegisterForReflectionBuildStep.java#L85-L87

But the method value(name).asBoolean() is not respecting the annotations default value, in such a way that it allways defaults to true. I'm using it to evaluate the new annotation parameter that I've created as sugested on this issue's history

So I had to reverse the predicate to allow a default to false and keep any backward compatibility.

It looks like a bug on jandex library
